routes/eth.go

```diff
@@ -0,0 +1,552 @@
+		return
+	}
+
+	// Calculate nanos purchased
+	totalWei := big.NewFloat(0).SetInt(ethTx.Total)
+	totalFees := big.NewFloat(0).SetInt(ethTx.Fees)
+	totalMinusFees := big.NewFloat(0).Sub(totalWei, totalFees)
+	totalEth := big.NewFloat(0).Quo(totalMinusFees, big.NewFloat(1e18))
+	nanosPurchased := fes.GetNanosFromETH(totalEth, feeBasisPoints)
+
+	bitcloutTxnHash, err := fes.SendSeedBitClout(pkBytes, nanosPurchased, true)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("SubmitETHTx: Error sending BitClout: %v", err))
+		return
+	}
+
+	res := SubmitETHTxResponse{
+		BitCloutTxnHash: bitcloutTxnHash.String(),
+	}
+	if err := json.NewEncoder(ww).Encode(res); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("SubmitETHTx: Problem encoding response: %v", err))
+		return
+	}
+}
+
+//
+// BlockCypher API
+//
+
+type BlockCypherTx struct {
+	BlockHeight int64                 `json:"block_height"`
+	BlockIndex  uint64                `json:"block_index"`
+	Hash        string                `json:"hash"`
+	Addresses   []string              `json:"addresses"`
+	Total       *big.Int              `json:"total"`
+	Fees        *big.Int              `json:"fees"`
+	Size        uint64                `json:"size"`
+	GasUsed     *big.Int              `json:"gas_used"`
+	GasPrice    *big.Int              `json:"gas_price"`
+	RelayedBy   string                `json:"relayed_by"`
+	Received    string                `json:"received"`
+	Ver         uint64                `json:"ver"`
+	DoubleSpend bool                  `json:"double_spend"`
+	VinSz       uint64                `json:"vin_sz"`
+	VoutSz      uint64                `json:"vout_sz"`
+	Inputs      []BlockCypherTxInput  `json:"inputs"`
+	Outputs     []BlockCypherTxOutput `json:"outputs"`
+}
+
+type BlockCypherTxInput struct {
+	Addresses []string `json:"addresses"`
+	Sequence  uint64   `json:"sequence"`
+}
+
+type BlockCypherTxOutput struct {
+	Addresses []string `json:"addresses"`
+	Value     *big.Int `json:"value"`
+}
+
+type BlockCypherCreateETHTxRequest struct {
+	Inputs  []BlockCypherTxInput  `json:"inputs"`
+	Outputs []BlockCypherTxOutput `json:"outputs"`
+}
+
+type BlockCypherCreateETHTxResponse struct {
+	Tx     BlockCypherTx `json:"tx"`
+	ToSign []string      `json:"tosign"`
+}
+
+func (fes *APIServer) BlockCypherCreateETHTx(ethAddress string, amount *big.Int) (*BlockCypherCreateETHTxResponse, error) {
+	URL := "https://api.blockcypher.com/v1/eth/main/txs/new"
+	if fes.Params.NetworkType == lib.NetworkType_TESTNET {
+		URL = "https://api.blockcypher.com/v1/beth/test/txs/new"
+	}
+
+	jsonData, err := json.Marshal(BlockCypherCreateETHTxRequest{
+		Inputs: []BlockCypherTxInput{
+			{
+				Addresses: []string{
+					ethAddress[2:], // Remove the 0x prefix
+				},
+			},
+		},
+		Outputs: []BlockCypherTxOutput{
+			{
+				Addresses: []string{
+					fes.Config.BuyBitCloutETHAddress[2:], // Remove the 0x prefix
+				},
+				Value: amount,
+			},
+		},
+	})
```

Comment:
Is there any way to prevent replace by fee? It would be bad if someone could buy clout with ETH and then broadcast a transaction to cancel their ETH txn: https://fixedfloat.com/en/blog/guides/replace-by-fee-ethereum#:~:text=What%20is%20the%20Replace%20by,after%20launch%20are%20considered%20unconfirmed.

Reply:
Hmmm, that's pretty bad. We probably need to wait for a block then